File: exercise/C#/day06/ElfWorkshop/ElfWorkshop.cs
namespace ElfWorkshop;

public class ElfWorkshop
{
    public List<string> TaskList { get; } = [];

    public void AddTask(string task)
    {
        if (task != null && task != "")
        {
            TaskList.Add(task);
        }
    }

    public string CompleteTask()
    {
        if (TaskList.Count > 0)
        {
            var task = TaskList[0];
            TaskList.RemoveAt(0);
            return task;
        }

        return null;
    }
}

File: exercise/kotlin/day06/src/main/kotlin/workshop/ElfWorkshop.kt
package workshop

class ElfWorkshop {
    val taskList: MutableList<String> = mutableListOf()

    fun getTaskList(): List<String> {
        return taskList
    }

    fun addTask(task: String?) {
        if (!task.isNullOrEmpty()) {
            taskList.add(task)
        }
    }

    fun completeTask(): String? {
        return if (taskList.isNotEmpty()) {
            taskList.removeAt(0)
        } else {
            null
        }
    }
}